### DISCUSSION TITLE
Universal Audio Apollo / ProTools / Blackhole /OBS

### DISCUSSION THREAD

**Discussion by jcbmusic** [ExistentialAudio/BlackHole#141] [General]
I run a twitch stream where I do live music production and songwriting. I just got a new computer with Catalina preinstalled, and as a consequence the JackRouter client that I used to use to route my ProTools audio, and interface audio into OBS no longer works. I'm wondering if there's a way to get input from my apollo while routing everything through BlackHole to OBS, so that I can record live in pro tools with the inputs from my apollo? Right now, I can open pro tools and get the session audio to run into OBS via Blackhole, but my inputs from my apollo in protools are sacrificed as a consequence of the new multi output device being the playback engine. In other words, I can show people my session, but I can’t record live.

To clarify, I’m referring to the instrument/mic inputs of my Apollo interface as non functional when Blackhole is running the show. When I open the pro tools session, and set the multi output device (Blackhole) to the playback engine, my ability to set an input from my Apollo as the input for an individual track in my session goes away. So I can hear my pro tools session output and my guitar (via Apollo Console and Blackhole) in OBS, but that guitar isn’t able to track in pro tools because the playback engine is the multi output device rather than the UA interface. When I set the Apollo to the playback engine in pro tools, it lets me use my Apollo inputs in my session again, but the audio from my session is no longer routed to OBS via Blackhole. I hope this makes a sense? I understand it’s hard to explain. I can try to capture a video of it if that might help.

Thanks to anybody who takes the time to try to help. Sincerely.

TLDR: I have three possible audio interfaces: Apollo, Blackhole, and the multi output device I’ve made using them. They all need to operate while running ProTools. I need to use my Apollo as the playback engine in pro tools so I have access to the Apollo inputs, and I need Blackhole to route my audio out to OBS.

**Comment by devinroth** [comment]
Your solution is to use an aggregate which combines all your audio devices into one. Then route the audio to the correct location.

**Comment by jcbmusic** [comment]
Hey Devin, thanks for your response. I made an aggregate in my audio/midi
setup with all of my devices in one place. I can open up a Pro Tools with
this aggregate as the playback engine, but many of my standard inputs are
still missing, and now my audio is no longer routed to OBS. Do you know if
there’s a tutorial available anywhere I could reference that shows how to
do this best?

**Comment by jcbmusic** [comment]
I’m unable to post it but I have a video of my problem if that would help with troubleshooting

**Comment by jcbmusic** [comment]
<img width="549" alt="Screen Shot 2020-05-09 at 12 06 23 PM" src="https://user-images.githubusercontent.com/65045681/81480209-cb299c80-91ed-11ea-8109-cef7cee0245e.png">
![Screen Shot 2020-05-09 at 12 05 19 PM](https://user-images.githubusercontent.com/65045681/81480211-cc5ac980-91ed-11ea-832c-2327d3f09f56.png)
<img width="668" alt="Screen Shot 2020-05-09 at 12 04 46 PM" src="https://user-images.githubusercontent.com/65045681/81480214-cebd2380-91ed-11ea-9cf9-4af58998383e.png">
<img width="667" alt="Screen Shot 2020-05-09 at 12 04 35 PM" src="https://user-images.githubusercontent.com/65045681/81480215-cf55ba00-91ed-11ea-8a45-0baa871381fb.png">
<img width="1488" alt="Screen Shot 2020-05-09 at 12 04 24 PM" src="https://user-images.githubusercontent.com/65045681/81480216-cf55ba00-91ed-11ea-832a-bb10a1e467b0.png">
<img width="746" alt="Screen Shot 2020-05-09 at 12 04 15 PM" src="https://user-images.githubusercontent.com/65045681/81480217-d1b81400-91ed-11ea-8f05-3276fed8c9e9.png">

**Comment by devinroth** [comment]
And a photo of your IO setup in PT please.

**Comment by jcbmusic** [comment]
<img width="993" alt="Screen Shot 2020-05-09 at 12 12 20 PM" src="https://user-images.githubusercontent.com/65045681/81480301-74709280-91ee-11ea-9a25-78c3e201eab8.png">
<img width="988" alt="Screen Shot 2020-05-09 at 12 12 33 PM" src="https://user-images.githubusercontent.com/65045681/81480302-763a5600-91ee-11ea-88c3-53d2d57e495b.png">

**Comment by devinroth** [comment]
BH is channel 33+ in the aggregate. But it doesn’t look like you’ve activated them in PT. Try clicking default in the IO setup.

**Comment by jcbmusic** [comment]
<img width="996" alt="Screen Shot 2020-05-09 at 12 22 39 PM" src="https://user-images.githubusercontent.com/65045681/81480517-ea292e00-91ef-11ea-8394-4be5dc0eeec3.png">
<img width="992" alt="Screen Shot 2020-05-09 at 12 23 08 PM" src="https://user-images.githubusercontent.com/65045681/81480518-ebf2f180-91ef-11ea-9063-6edf5d60d069.png">

**Comment by jcbmusic** [comment]
I will literally pay you money if you can help me succeed in getting this working. I'm at a loss here.

**Comment by devinroth** [comment]
What version of pt are you on?

Devin

**Comment by jcbmusic** [comment]
When I got the new computer with Catalina preinstalled, I traded my PT 11
perpetual for the subscription based pro tools, so I’m running whatever the
most recent patch of that is.

Could I be doing something wrong with my aggregate device? Is there a
specific channel or way I’m supposed to route audio in pro tools to get it
to send to Blackhole?

**Comment by devinroth** [comment]
Okay one more thing. Show me how to have the tracks setup in PT.

**Comment by devinroth** [comment]
Have you tried with an aggregate with only your Apollo? If that doesn't work I can add the blame Apple tag to this.

**Comment by jcbmusic** [comment]
I mean, the aggregate with my apollo and Blackhole functions and I can record with it and everything. Its just not routing my audio to OBS

**Comment by jcbmusic** [comment]
<img width="1764" alt="Screen Shot 2020-05-11 at 12 58 12 PM" src="https://user-images.githubusercontent.com/65045681/81594926-42deff00-9387-11ea-9d1f-341310b05ae5.png">

**Comment by devinroth** [comment]
Does it work with something other than OBS? Like Zoom?

**Comment by devinroth** [comment]
Are you able to get OBS to work with just the Apollo?

**Comment by jcbmusic** [comment]
I’ll have to try it with zoom, I’ve never used it before though. I’ve
managed to get it so Blackhole can route audio like Spotify etc to OBS, my
Apollo works to route my mic to OBS, but the only way I can get my PT
session audio to go to OBS via Blackhole is if I use Blackhole or Multi
Output Device as the playback engine, which removes my ability to record
live on stream. Which is what my whole stream hinges on. For some reason,
my aggregate device has never passed audio to OBS when routed to an audio
input.

**Comment by jcbmusic** [comment]
So I can record in ProTools with the aggregate device as the playback
engine, but it doesn’t route to OBS

**Comment by devinroth** [comment]
Okay so don’t worry about zoom. 

But have you tested an aggregate with just your Apollo as the only device?

**Comment by jcbmusic** [comment]
Just did, no change. Still no output to OBS on aggregate device channel.

**Comment by devinroth** [comment]
So this officially earns the Blame Apple tag. 

What you can try is seeing if different permutations of audio devices in the aggregate solve the problem. Try adding in the build it input/output too. Basically try every possible way to see if you get it to work. 

Beyond that I'm afraid I'm out of ideas.

**Comment by jcbmusic** [comment]
Dang :/ yeah I’ve tried every last thing short of a hardware workaround,
which may still be an option. I.e run a headphone out into a line in on my
Apollo or something like that. Well your software is really wonderful for
just about anything else anyone would wanna do with it. Thanks for all your
help, sincerely. Do you think you can unofficially/ off the record
recommend any other software or solution that could work for me? Maybe just
wait for a future update?

**Comment by darktrunk** [comment]
Does this link help
https://www.youtube.com/watch?v=miexiUIusts&feature=share

**Comment by jcbmusic** [comment]
Thanks for the message man. Unfortunately, due to PT not being able to set
different input and output devices, this video does not help me :/ but
otherwise, this video is a BEAUTIFUL resource. If I ever figure it out,
I’ll make a PT tutorial modeled after that one for sure

**Comment by darktrunk** [comment]
Hey
Really sorry to hear. I got caught up with making aggregate drivers and was clicking the wrong option in audio/midi.....watched this vid and it worked perfectly. Didn’t know that about PT.....thus far this method has worked on ableton and presonus. Best of luck.


Sent from Yahoo Mail for iPhone


On Friday, May 15, 2020, 06:24, jcbmusic <notifications@github.com> wrote:



Thanks for the message man. Unfortunately, due to PT not being able to set
different input and output devices, this video does not help me :/ but
otherwise, this video is a BEAUTIFUL resource. If I ever figure it out,
I’ll make a PT tutorial modeled after that one for sure

**Comment by coronin** [comment]
install blackhole broke my OBS, and uninstall solved the OBS audio recording issue -- I didn't change anything with OBS

**Comment by pkviet** [comment]
the reason it might not work with obs is obs on mac only supports audio devices with up to 8 audio channels. There's a PR lifting that limitation though:
https://github.com/obsproject/obs-studio/pull/2889
 @jcbmusic you might want to try this PR 
(aside from this PR, i have an obs fork which supports up to 16 channels though https://github.com/pkviet/obs-studio/releases ; it might do the job until the PR is reviewed and merged)

**Comment by jbuecker** [comment]
Also realize, its Quite easy to recompile and Build a BLACKHOLE to run at fewer channels (You can build an 8 channel or a two channel) or whatever, Check out the WIKI HTH, ~JB

**Comment by PremiumJoeCompositions** [comment]
I know this is a bit late, but maybe it will help someone.  

I'm operating a mid 2011 iMac OS 10.11.6 and Pro Tools 12

**Audio/Midi Setup:** Create Aggregate Device, Select interface first then Blackhole. I also set the Aggregate Device to for sound input and output by right-clicking on it (Don't know if it's necessary, but just do it).
**System Preferences: Sound:** Set Aggregate Device for the Output and Input (Don't know if it's necessary, but just do it). Mute Internal Speakers.
**ProTools:** Set Playback Engine to Aggregate Device, I set my I/O to default which means my output has 1-2 and 3-4. Create AUX track going out to Master 1-2. AUX has a send going out to 3-4. Route all tracks to AUX track.
**OBS:** Mic 1 is Blackhole 16ch (or 2ch if you have that version), Mic 2 is my Interface.

This works for me. Hope this saves someone a lot of time.

Regards,

Joseph
